File: source_code/project4_flutter/lib/features/property_detail/models/custom_paging.dart
import 'package:json_annotation/json_annotation.dart';
part  'custom_paging.g.dart';
@JsonSerializable()
class CustomPaging {
  final int status;
  final String message;
  final int totalPages;
  final int currentPage;
  final int pageSize;
  final int totalCount;
  final bool hasPrevious;
  final bool hasNext;
  // dynamic data;

  CustomPaging({ required this.status,
    required this.message,
    required this.currentPage,
    required this.totalPages,
    required this.pageSize,
    required this.totalCount,
    required this.hasPrevious,
    required this.hasNext,
    // required this.data
  });

  factory CustomPaging.fromJson(Map<String, dynamic> json) => _$CustomPagingFromJson(json);
  Map<String, dynamic> toJson() => _$CustomPagingToJson(this) ;
}

File: source_code/project4_spring/chat/src/main/java/com/service/chat/dto/CustomPaging.java
package com.service.chat.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@Data
@NoArgsConstructor
public class CustomPaging {
    private int status;

    private String message;

    private int currentPage;

    private int totalPages;

    private int pageSize;

    private long totalCount;

    private boolean hasPrevious;

    private boolean hasNext;

    public boolean isHasPrevious() {
        return currentPage > 0;
    }

    public boolean isHasNext() {
        return currentPage + 1 < totalPages;
    }

    private Object data;
}